src/internal/hooks/use-virtual/__tests__/use-virtual-offset.test.tsx

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useRef } from 'react';
+import { render } from '@testing-library/react';
+
+import { useVirtual } from '../index';
+
+// Mock the vendor react-virtual to avoid ES module issues
+jest.mock('../../../vendor/react-virtual', () => ({
+  useVirtual: jest.fn(),
+}));
+
+import * as reactVirtualModule from '../../../vendor/react-virtual';
+
+const mockUseVirtual = reactVirtualModule.useVirtual as jest.MockedFunction<any>;
+
+interface TestItem {
+  id: string;
+}
+
+const TestComponent = ({
+  items,
+  firstItemSticky,
+  itemOverlap,
+  onResult,
+}: {
+  items: TestItem[];
+  firstItemSticky?: boolean;
+  itemOverlap?: number;
+  onResult: (result: ReturnType<typeof useVirtual<TestItem>>) => void;
+}) => {
+  const parentRef = useRef<HTMLDivElement>(null);
+  const result = useVirtual({
+    items,
+    parentRef,
+    estimateSize: () => 50,
+    firstItemSticky,
+    itemOverlap,
+  });
+
+  onResult(result);
+
+  return (
+    <div ref={parentRef} style={{ height: '500px', overflow: 'auto' }}>
+      {result.virtualItems.map(item => (
+        <div key={item.index} style={{ height: item.size }}>
+          Item {item.index}
+        </div>
+      ))}
+    </div>
+  );
+};
+
+describe('useVirtual', () => {
+  const mockMeasureRef = jest.fn();
+  const mockScrollToIndex = jest.fn();
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('adjusts totalSize with itemOverlap', () => {
+    const items: TestItem[] = [{ id: '1' }, { id: '2' }, { id: '3' }];
+
+    // Mock react-virtual to return 3 items with size 50 each, total 150
+    mockUseVirtual.mockReturnValue({
+      virtualItems: [
+        { index: 0, size: 50, start: 0, end: 50, measureRef: mockMeasureRef },
+        { index: 1, size: 50, start: 50, end: 100, measureRef: mockMeasureRef },
+        { index: 2, size: 50, start: 100, end: 150, measureRef: mockMeasureRef },
+      ],
+      totalSize: 150,
+      scrollToIndex: mockScrollToIndex,
+    });
+
+    let capturedResult: ReturnType<typeof useVirtual<TestItem>> | null = null;
+    render(<TestComponent items={items} itemOverlap={1} onResult={result => (capturedResult = result)} />);
+
+    expect(capturedResult).not.toBeNull();
+    // totalSize = 150 - (3 items * 1 overlap) = 147
+    expect(capturedResult!.totalSize).toBe(147);
+  });
+
+  test('adjusts totalSize with no itemOverlap', () => {
```

Comment:
nit: the test name is not quite correct as no total size adjustment is needed. This can be renamed to "computes totalSize with itemOverlap=0" (and update the above test accordingly).